pkg/controller/handlers/auditlogexport/auditlogexport.go

```diff
@@ -0,0 +1,240 @@
+			CallType:                  export.Spec.Filters.CallTypes,
+			CallIdentifier:            export.Spec.Filters.CallIdentifiers,
+			SessionID:                 export.Spec.Filters.SessionIDs,
+			ClientName:                export.Spec.Filters.ClientNames,
+			ClientVersion:             export.Spec.Filters.ClientVersions,
+			ResponseStatus:            export.Spec.Filters.ResponseStatuses,
+			ClientIP:                  export.Spec.Filters.ClientIPs,
+			Query:                     export.Spec.Filters.Query,
+			Limit:                     batchSize,
+			Offset:                    offset,
+			WithRequestAndResponse:    export.Spec.WithRequestAndResponse,
+		}
+
+		// Get batch of logs from gateway
+		logs, total, err := h.gatewayClient.GetMCPAuditLogs(ctx, opts)
+		if err != nil {
+			return 0, 0, fmt.Errorf("failed to get audit logs batch %d: %w", batchNumber, err)
+		}
+
+		totalRecords = total
+
+		// If no logs in this batch, we're done
+		if len(logs) == 0 {
+			break
+		}
+
+		// Convert logs to the desired format
+		batchData, err := h.formatLogs(logs)
+		if err != nil {
+			return 0, 0, fmt.Errorf("failed to format logs batch %d: %w", batchNumber, err)
+		}
+
+		_, err = pw.Write(batchData)
+		if err != nil {
+			return 0, 0, fmt.Errorf("failed to write to pipe: %w", err)
+		}
+
+		totalSize += int64(len(batchData))
+		offset += len(logs)
+		batchNumber++
+	}
+
+	if err := pw.Close(); err != nil {
+		return totalRecords, totalSize, fmt.Errorf("failed to close pipe: %w", err)
+	}
+
+	// Wait for upload to complete
+	if err := <-uploadErrCh; err != nil {
+		return totalRecords, totalSize, fmt.Errorf("upload failed: %w", err)
+	}
+
+	return totalRecords, totalSize, nil
+}
+
+func (h *Handler) formatLogs(logs []gatewaytypes.MCPAuditLog) ([]byte, error) {
+	lines := make([]string, 0, len(logs))
+
+	// Convert each log to a JSON line
+	for _, log := range logs {
+		apiLog := gatewaytypes.ConvertMCPAuditLog(log)
+		jsonBytes, err := json.Marshal(apiLog)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal log entry: %w", err)
+		}
+		lines = append(lines, string(jsonBytes))
+	}
+
+	// Join with newlines and add a final newline
+	result := strings.Join(lines, "\n")
+	if len(lines) > 0 {
+		result += "\n"
+	}
+
+	return []byte(result), nil
+}
+
+func (h *Handler) generateExportPath(export *v1.AuditLogExport) string {
+	now := time.Now()
+	timestamp := now.Format(time.RFC3339)
+	filename := fmt.Sprintf("%s-%s.jsonl", export.Spec.Name, timestamp)
+
+	// Use keyPrefix if provided, otherwise use default date-based prefix
+	keyPrefix := export.Spec.KeyPrefix
+	if keyPrefix == "" {
+		// Generate default prefix with year/month/day
+		keyPrefix = fmt.Sprintf("mcp-audit-logs/%04d/%02d/%02d", now.Year(), now.Month(), now.Day())
+	}
+
+	// Ensure keyPrefix ends with / if it's not empty
+	if keyPrefix != "" && !strings.HasSuffix(keyPrefix, "/") {
+		keyPrefix += "/"
+	}
+
+	return keyPrefix + filename
```

Comment:
Do we want to sanitize this to remove things like `..`? I don't know if path traversal is actually a risk since we are uploading to blob storage, but figured I should ask.